### ISSUE TITLE
A dashboard for monitoring user's progress.

### ISSUE THREAD

**Issue by AkshayAgarwal007** [Vigneshsekar/Tux4ME#30]
A dashboard feature needs to be implemented. The data about the user's performance depending on his score, the time has takes to complete a level should be stored and logged into a dashboard. For the web app deployed on Azure Azure SQL database could be used or HTML Web storage to store the data locally within the user's browser. In case of the hybrid app interfacing it with IndexedDB or MongoDB would be a good idea.

**Comment by Vigneshsekar** [comment]
Thanks for your inputs @AkshayAgarwal007. We're currently working on the same.

**Comment by kvk30** [comment]
I have doubt, do we really require a dash board for 7-14 years kids? Here we are helping them to enhance there thinking ability. What @Vigneshsekar  implemented is correct way. Sessions are better than dash board. No offence for your idea @AkshayAgarwal007. Correct me, If I am wrong.

**Comment by Vigneshsekar** [comment]
I agree with you @kvk30. But @AkshayAgarwal007 is speaking in long terms. We should look into that and perhaps the same can be implemented during this summer. For now, may be we can show the results to them in a better way. With some charts or so ?

**Comment by AkshayAgarwal007** [comment]
@kvk30 , I am telling this based on the survey I have done, I have talked with several parents and non profit organisations working for educating young kids about what features they would want in an educational game and they all at sometime have used TuxType and TuxMath. They all want a dashboard feature so they can monitor the kid's progress, because for kid's of the age group 7-14, parents want a feature to keep a check on the progress of their kids. Also from the perspective of a game you bind the interest of the users by introducing badges, coins etc and that would be implemented using the dashboard feature. Take the case of angry bird, a user would keep playing a level until he earns all the three stars and take pride in viewing them all later.

The intent of the game and the strategy by which you bind the users are two different things. Moreover in this case the dashboard feature is a necessity for NPO's and parents.

In any case, I can only suggest features, end of the day it's @Vigneshsekar 's  call that what he wants to get implemented,

**Comment by karthikeyan5** [comment]
I agree with @AkshayAgarwal007 .  But seeing to the previous comments of @Vigneshsekar , he said that he is working on the same. So it would be better if we could give more ideas and inputs on the dashboard feature. I am currently making a simple UI representation of the dashboard. Will send to @Vigneshsekar for more inputs.

**Comment by Vigneshsekar** [comment]
Its not only my call @AkshayAgarwal007. If you make any good suggestions and if you're able to justify it, we will definitely consider it.

**Comment by AkshayAgarwal007** [comment]
@Vigneshsekar Vigneshsekar Noted.
@karthikeyan5 Yes you are right, I've made a responsive UI for the dashboard using Bootstrap, I think we should keep it simple and elegant but again if we want to give it game like elements it should be thought of. Moreover making it responsive is necessary as we are to wrap this game for mobile devices. 

Also apart from the UI we should think of adding more game based incentives like badges , coins in the future, so the design of the dashboard needs to be be modular so that it can be easily extended based on the additions in the game.

![dash1](https://cloud.githubusercontent.com/assets/8170214/13915484/04f059aa-ef7b-11e5-839b-984de820f063.png)
![dash2](https://cloud.githubusercontent.com/assets/8170214/13915485/0509e938-ef7b-11e5-93ab-a51f824b2de2.png)

**Comment by kvk30** [comment]
@AkshayAgarwal007 Have tried to implement dashboard on Tux4me.

**Comment by AkshayAgarwal007** [comment]
@kvk30 I would like to see your work. Though I would suggest you not to push it too hard as building a good one would take a lot of research and again it should be modular so that the underlying database could be changed very easily and also binding the same to any new game should be easy.

**Comment by kvk30** [comment]
@AkshayAgarwal007 why do you want to see my work? By the way, I am not interested in creating an dashboard for Tux4me. you had commented and showed some images regarding dashboard. I thought you had a prototype to build dashboard and also a working model. I felt guilty for objecting your idea, like to help you in this issue. :)

**Comment by AkshayAgarwal007** [comment]
@kvk30  Sorry, I didn't get what you were saying before, thought you have done something. It isn't a simple dashboard that just logs the score up and the UI mockups that you see are very basic. This is a big chunk of work if one makes a more generalized one. Presently I'm designing the database schema. You need not be guilty, these discussions always helps in knowing various perceptions, not everyone whom I surveyed wanted a dashboard but most of them did, it completely depends on the perspective from which you are looking at these games.
I appreciate your help :) but would wait for this till the results. Till then #49 and #39 would be something really great to do as it would help anyone who would be working on this project during GSoC.
